sky/engine/core/script/dart_init.cc

```diff
@@ -341,6 +328,74 @@ void* _DartSymbolLookup(const char* symbol_name) {
   return DartLookupSymbolInLibrary(symbol_name, nullptr);
 }
 
+#elif OS(ANDROID)
+
+// Describes an asset file that holds a part of the precompiled snapshot.
+struct SymbolAsset {
+  SymbolAsset(const char* file, bool exec)
+      : file_name(file), is_executable(exec) {};
+  const char* file_name;
+  bool is_executable;
+};
+
+static const std::map<std::string, SymbolAsset> g_symbol_asset_map = {
+  {kDartVmIsolateSnapshotBufferName,
+   SymbolAsset("snapshot_aot_vmisolate", false)},
+  {kDartIsolateSnapshotBufferName, SymbolAsset("snapshot_aot_isolate", false)},
+  {kInstructionsSnapshotName, SymbolAsset("snapshot_aot_instr", true)},
+  {kDataSnapshotName, SymbolAsset("snapshot_aot_rodata", false)},
+};
+
+static std::map<std::string, void*> g_symbol_cache;
+
+// Resolve a precompiled snapshot symbol by mapping the corresponding asset
+// file into memory.
+void* _DartSymbolLookup(const char* symbol_name) {
+  auto cache_it = g_symbol_cache.find(symbol_name);
```

Comment:
Do we need this cache?  Why not just store the void* inside the SymbolAsset struct?